clientd3d/gameuser.c

```diff
@@ -147,13 +148,19 @@ void GotObjectContents(ID object_id, list_type contents)
 	 GameMessagePrintf(GetString(hInst, IDS_EMPTY), LookupNameRsc(r->obj.name_res));
       return;
    }
+   //Separate contents into number items and other items and alpha sort
+   for (l = contents; l != NULL; l = l->next)
+   {
+      sorted_list = list_add_sorted_item(sorted_list, (l->data), CompareObjectNameAndNumber);
+   }
```

Comment:
Here I've updated the method so that the compare function of the insertion sort so that the comparison does the stacking of number items and also the alphabetical sort of number items together and then not number items together.  As suggested by Zaphod.  That comment appears to have disappeared with the edits.

Reply:
Also we are no longer casting the data to object_node here because list_add_sorted_item passes off a void to the comparison function anyhow and forcing the issue in the comparison function call looked like it would require changing list_add_sorted_item which is used elsewhere for other tasks.